examples/examples_test.go

```diff
@@ -0,0 +1,261 @@
+// +build examples
+
+/*
+Copyright 2020 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"os"
+	"os/exec"
+
+	"github.com/tektoncd/pipeline/pkg/names"
+	knativetest "knative.dev/pkg/test"
+)
+
+const (
+	timeoutSeconds = 600
+	sleepBetween   = 10
+
+	// we may want to consider either not running examples that require registry access
+	// or doing something more sophisticated to inject the right registry in when folks
+	// are executing the examples
+	horribleHardCodedRegistry = "gcr.io/christiewilson-catfactory"
```

Comment:
nit: Why not running a local registry (in the test namespace) ? (in any case, it would be a follow-up)

Reply:
could be a good follow up! unless the test is deploying to a different namespace 🤔